Documentation/network/egress-gateway.rst

```diff
@@ -126,6 +126,11 @@ interface as follow:
 
             install-egress-gateway-routes: true
 
+.. warning::
+
+  The ``install-egress-gateway-routes`` flag or ``egressGateway.installRoutes`` Helm value is only 
+  needed on EKS's ENI mode.
+
```

Comment:
Apologies, I think I went too fast on my first review. Coming back to this PR, I realise that it reads a bit awkward to have users read this whole subsection on `egressGateway.installRoutes`, only to tell them at the end that, if using anything else than EKS, they should in fact have skipped this paragraph entirely.

Could we instead have a lower-level heading to indicate it from the start? Something like:

```
Compatibility with cloud environments
-------------------------------------

EKS's ENI mode
~~~~~~~~~~~~~~

Based on the specific configuration of the cloud provider and network interfaces
it is possible that traffic leaves a node from the wrong interface. This happens in particular on EKS in ENI mode.

To work around this issue...
```

and then drop the warning. What do you think?

Sorry for the additional work, and do let me know if I can help with squashing these commits (`git rebase -i master update-egress-doc` should help).

Reply:
Sorry to hear you've been spending so much time on this!

Here's how I would do it, in case you want to do the same on your side. If this still does not work (or if you don't want to spend more time on it), let me know and I'll squash and force-push on your branch tomorrow.

So I pull the `master` branch, then I checkout out your PR locally:

![image](https://user-images.githubusercontent.com/17001771/218573232-c839b956-f99f-47e1-8980-0d99552fe05c.png)

I want to rebase interactively. I run `git rebase -i master`. An editor opens (Neovim in my case) to let me choose what I want to do with the commits, see the help message in grey:

![image](https://user-images.githubusercontent.com/17001771/218573380-736f2715-1ffb-493b-a760-1124b58acc7f.png)

So I pick the first commit (I don't change that line), but set everything else to `s`, for `squash`:

![image](https://user-images.githubusercontent.com/17001771/218573504-7b33b50e-63d3-4b0d-8d38-1d55fc33ade5.png)

I save and quit and Git attempts to rebase everything. Bad luck, we get conflicts (not sure why because I didn't move commits around, I guess this is due to the merge commits).

![image](https://user-images.githubusercontent.com/17001771/218573800-bfc73075-fd6a-4911-b7d7-af0b735d6072.png)

I open the file to solve the conflict, here by deleting the portion in red:

![image](https://user-images.githubusercontent.com/17001771/218573896-790b3ffe-a40e-4b22-ad19-e305d2d82295.png)

I add the change (`git add`) and continue, `git rebase --continue`:

![image](https://user-images.githubusercontent.com/17001771/218575096-428160c0-964a-4fcd-9576-99b466fd3e73.png)
![image](https://user-images.githubusercontent.com/17001771/218575216-a6922c00-b78d-4004-bd67-1b5917ffa2ae.png)

Now Git can squash. It asks me what the resulting commit log should be. The editor opens:

![image](https://user-images.githubusercontent.com/17001771/218574426-8de7db20-2c75-4d5c-9b8a-cdd83edb4299.png)

I keep only the relevant part, edits the commit object, ideally add a commit description:

![image](https://user-images.githubusercontent.com/17001771/218574587-aaef5634-cf7d-4aac-bd7b-b286f71595a0.png)

Then I save and exit the editor. At this point I had a second conflict, which I solved with the same steps (edit file, `git add`, `git rebase --continue`). After that conflict, the editor opens again to edit the message resulting from the squash of the last commits.

After that, the rebase operation terminates, and I only have a single commit left with the changes. All good now! 

![image](https://user-images.githubusercontent.com/17001771/218574882-3f6cd3b0-e0c6-4e82-9a04-b9bc1a96dae3.png)

Again, just let me know how this goes for you.